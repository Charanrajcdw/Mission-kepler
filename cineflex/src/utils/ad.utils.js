import ad1 from "../assets/advertisements/small/small-ad1.png";
import ad2 from "../assets/advertisements/small/small-ad2.png";

const ads = [ad1, ad2];

export const getRandomAd = () => {
  return ads[Math.floor(Math.random() * ads.length)];
};
