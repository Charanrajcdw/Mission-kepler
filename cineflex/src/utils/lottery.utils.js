import { LOTTERY } from "../constants";

export const checkLotteryNo = (enteredNumber) => {
  const number = +enteredNumber;
  if (isNaN(number) || enteredNumber.length != 10) return LOTTERY.invalidNumber;
  const lastDigit = +enteredNumber.slice(-1);
  if (lastDigit % 2 !== 0) return LOTTERY.unluckyMessage;
};
