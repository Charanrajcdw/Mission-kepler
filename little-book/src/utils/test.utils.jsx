import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import ThemeContextProvider from "../contexts/theme.context";
import blogSlice from "../store/slices/blogSlice";
import { THEME } from "../constants";

const customRender = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
    ...options
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const customCreateSnapshot = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return renderer.create(<ProviderWrapper>{ui}</ProviderWrapper>).toJSON();
};

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render, customCreateSnapshot as createSnapshot };
