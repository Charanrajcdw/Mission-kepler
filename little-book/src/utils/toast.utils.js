import { toast } from "react-toastify";

export const showSuccessToast = (message, currentTheme) => {
  toast.success(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};
