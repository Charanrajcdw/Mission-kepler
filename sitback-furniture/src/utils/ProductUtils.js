export const transformToIndianRupee = (price) => {
  price = parseInt(price);
  return price.toLocaleString("en-IN");
};

export const getGuaranteeMessage = (guarantee) => {
  return guarantee === 1 ? `1 YEAR GUARANTEE` : `${guarantee} YEARS GUARANTEE`;
};
