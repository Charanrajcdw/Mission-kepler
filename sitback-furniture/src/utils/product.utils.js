import { CART } from "../constants/constants";

export const transformToIndianRupee = (price) => {
  price = +price;
  return price ? price.toLocaleString("en-IN", { maximumFractionDigits: 2 }) : 0;
};

export const getGuaranteeMessage = (guarantee) => {
  return guarantee === 1 ? `1 YEAR GUARANTEE` : `${guarantee} YEARS GUARANTEE`;
};

export const calculateTotalPrice = (items) => {
  const price = items.reduce((total, product) => total + parseInt(product.price) * product.quantity, 0);
  return transformToIndianRupee(price);
};

export const getItemsFromLocalStorage = (key) => {
  return JSON.parse(localStorage.getItem(key)) ?? [];
};

export const setItemsToLocalStorage = (key, items) => {
  localStorage.setItem(key, JSON.stringify(items));
};

export const addToCart = (product, quantity) => {
  const cartItems = getItemsFromLocalStorage(CART.cart);
  const productIndex = cartItems.findIndex((item) => item.id === product.id);
  if (productIndex >= 0) {
    cartItems[productIndex].quantity += quantity;
    if (cartItems[productIndex].quantity === 0) {
      cartItems.splice(productIndex, 1);
    }
  } else {
    cartItems.push({ ...product, quantity: quantity });
  }
  setItemsToLocalStorage(CART.cart, cartItems);
  return cartItems;
};

export const addToWishlist = (product) => {
  const wishlistItems = getItemsFromLocalStorage(CART.wishlist);
  const productIndex = wishlistItems.findIndex((item) => item.id === product.id);
  if (productIndex < 0) {
    wishlistItems.push(product);
  }
  setItemsToLocalStorage(CART.wishlist, wishlistItems);
  return wishlistItems;
};

export const removeFromWishlist = (product) => {
  const wishlistItems = getItemsFromLocalStorage(CART.wishlist);
  const newWishlistItems = wishlistItems.filter((item) => item.id !== product.id);
  setItemsToLocalStorage(CART.wishlist, newWishlistItems);
  return newWishlistItems;
};
