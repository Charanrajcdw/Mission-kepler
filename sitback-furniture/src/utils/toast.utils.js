import { toast } from "react-toastify";

export const showSuccessToast = (message) => {
  toast.success(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: "light",
  });
};

export const showWarningToast = (message) => {
  toast.warn(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: "light",
  });
};
