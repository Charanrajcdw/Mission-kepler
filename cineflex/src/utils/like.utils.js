import { LIKES } from "../constants";
import { localStorageHelper } from "./localStorage.utils";

const { get, set } = localStorageHelper;

export const storeLikedMovie = (id) => {
  const likedMovies = get(LIKES.key) ?? [];
  if (!likedMovies.includes(id)) likedMovies.push(id);
  set(LIKES.key, likedMovies);
};
