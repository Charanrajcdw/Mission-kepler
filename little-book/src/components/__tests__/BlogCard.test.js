import { render, screen, createSnapshot, userEvent } from "../../utils/test.utils.jsx";
import { BlogCard } from "..";

describe("BlogCard", () => {
  it("should render Blog Card", () => {
    const blogData = {
      title: "Blog title",
      type: "Local",
      details: "Blog details",
    };
    render(<BlogCard blog={blogData} isSelected={true} blogChangeHandler={() => {}} />);
    const titleElement = screen.getByText(blogData.title);
    const detailsElement = screen.getByText(blogData.details);
    const typeElement = screen.getByText(blogData.type);
    expect(titleElement).toBeInTheDocument();
    expect(detailsElement).toBeInTheDocument();
    expect(typeElement).toBeInTheDocument();
  });

  it("should match Blog Card snapshot", () => {
    const blogData = {
      title: "Blog title",
      type: "Local",
      details: "Blog details",
    };
    const blogcard = createSnapshot(<BlogCard blog={blogData} isSelected={true} blogChangeHandler={() => {}} />);
    expect(blogcard).toMatchSnapshot();
  });

  it("should update current blog", async () => {
    const user = userEvent.setup();
    const blogData = {
      title: "Blog title",
      type: "Local",
      details: "Blog details",
    };
    const blogChangeHandler = jest.fn();
    render(<BlogCard blog={blogData} isSelected={true} blogChangeHandler={blogChangeHandler} />);
    const titleElement = screen.getByText(blogData.title);
    const detailsElement = screen.getByText(blogData.details);
    await user.click(titleElement);
    await user.click(detailsElement);
    expect(blogChangeHandler).toHaveBeenCalledTimes(2);
  });
});
