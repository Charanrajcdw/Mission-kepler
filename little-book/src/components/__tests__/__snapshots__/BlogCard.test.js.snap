// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard should match Blog Card snapshot 1`] = `
<div
  className="blogCard selected"
  onClick={[Function]}
>
  <h6
    className="blogTitle"
  >
    Blog title
  </h6>
  <p
    className="blogType"
  >
    Local
  </p>
  <p
    className="blogDetails"
  >
    Blog details
  </p>
</div>
`;
